### ISSUE TITLE
Patch upgrade bypassed requirement to validate minor upgrade

### ISSUE THREAD

**Issue by c9mbundy** [umbraco/Umbraco.Cloud.Issues#171]
I had intentionally held my baseline-master and baseline-child sites at Umbraco-Forms 7.1.3 because of bugs in both 7.2.0 and 7.3.0 minor releases that I knew would seriously impact my clients, as they were similar to bugs in 7.1.1 and 7.1.2 patch upgrades that had previously caused me problems.

I was able to do this because minor upgrades require testing and approval on a dev environment of the baseline-master, before pushing to the live environment and then deploying to child sites.

I was intending to test the latest 7.3.1 release on the baseline master's dev environment and satisfy myself that the latest patch upgrade had fixed the issues in the previous minor upgrades, and were not going to cause a problem for my clients, before deploying to child sites. 

Umbraco-Cloud support had previously advised that this was a valid approach, and that I could hold any minor upgrade for as long as I needed to, and that a patch upgrade would not be applied without the minor upgrade being previously approved and deployed.

However, on Tuesday, Umbraco pushed patch upgrades that bypassed the requirement to approve minor upgrades, and all sites - baseline master and child - have been forced from 7.1.3 directly to 7.3.1 - i.e. pushing 2 minor upgrades without my approval.

I guess that the question is how did a patch force 2 minor upgrades to all sites? 

This would appear to be a bug in the deploy process, as a patch has effectively circumvented the requirement of approval for minor upgrades.

Perhaps consideration should also be given to allowing more control over the upgrade process, as my attempt to stop this was as a result of 2 earlier patch upgrades auto-magically breaking my sites, and this would appear to be a problem with the current process.

**Comment by sitereactor** [comment]
Hi @c9mbundy thanks for your detailed write-up! We'll need to look into what happened here, before I can give you a good answer. But it sounds to me like our pre-checks missed something, which means the patch was pushed when it should not have been.

We'll get back to this thread when we know more.

**Comment by mikkelhm** [comment]
Hi @c9mbundy - First sorry for the long wait - Thanks for reporting.

Could I get you to write me a DM here or an email at mikkel-at-umbraco.com, with the name of project in question? 

It sure sounds like you did everything according to plan, and the upgrade shouldn't have been rolled out. Normally a minor upgrade should never be able to go from the baseline to its children without using the "Update baseline child" feature, which is a user invoked action, and something that wont be run from the autoupgrade process. 

Anyways I'd like to know the baseline projects name, and also which children you saw the issue on ( I might be able to figure out, but would like it stated so I dont get it wrong)

Kind regards - Mikkel

**Comment by mikkelhm** [comment]
Hi @c9mbundy 

I just looked a bit further into the code, and found the bug - it is totally on our side. For some reason when we are doing patch upgrades of forms, and your site is on Umbraco 7, you will just get the latest minor version of forms, not just the patch.
It was fixed fine for sites running Umbraco version 8. I've applied the same check for sites on Umbraco version 7, so this should never happen again. 

Sorry for the trouble it has given you and your clients.
